From 0c717b9ee5857da2f81a6be59ca25063d79658f2 Mon Sep 17 00:00:00 2001
From: Alin Jerpelea <alin.jerpelea@sony.com>
Date: Tue, 30 Jan 2018 03:15:02 +0900
Subject: [PATCH] hal: enable audio hal on sdm660

the sdm660 SOC can use the same HAL as the other QCOM SOCs

Test: manual - build, boots and audio works

Change-Id: I0edd5fa2c67eb7a96a44e907060dcbb273e983ac
Signed-off-by: Alin Jerpelea <alin.jerpelea@sony.com>
---
 Android.mk            | 2 +-
 hal/Android.mk        | 2 +-
 hal/msm8916/hw_info.c | 9 +++++++--
 3 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/Android.mk b/Android.mk
index 65bf578..371486a 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,6 +1,6 @@
 # TODO:  Find a better way to separate build configs for ADP vs non-ADP devices
 ifneq ($(TARGET_BOARD_AUTO),true)
-  ifneq ($(filter msm8960 msm8226 msm8x26 msm8x84 msm8084 msm8992 msm8994 msm8996 msm8909 msm8952 msm8998,$(TARGET_BOARD_PLATFORM)),)
+  ifneq ($(filter msm8960 msm8226 msm8x26 msm8x84 msm8084 msm8992 msm8994 msm8996 msm8909 msm8952 msm8998 sdm660,$(TARGET_BOARD_PLATFORM)),)
 
     MY_LOCAL_PATH := $(call my-dir)
 
diff --git a/hal/Android.mk b/hal/Android.mk
index 923879c..1f3c02d 100644
--- a/hal/Android.mk
+++ b/hal/Android.mk
@@ -49,7 +49,7 @@ ifneq ($(filter msm8998,$(TARGET_BOARD_PLATFORM)),)
 endif
 endif
 
-ifneq ($(filter msm8916 msm8909 msm8952,$(TARGET_BOARD_PLATFORM)),)
+ifneq ($(filter msm8916 msm8909 msm8952 sdm660,$(TARGET_BOARD_PLATFORM)),)
   AUDIO_PLATFORM = msm8916
   LOCAL_CFLAGS := -DPLATFORM_MSM8916
 ifneq ($(filter msm8909,$(TARGET_BOARD_PLATFORM)),)
diff --git a/hal/msm8916/hw_info.c b/hal/msm8916/hw_info.c
index a4f3943..e2cc44e 100644
--- a/hal/msm8916/hw_info.c
+++ b/hal/msm8916/hw_info.c
@@ -54,8 +54,12 @@ static void update_hardware_info_8x16(struct hardware_info *hw_info, const char
         strlcpy(hw_info->name, "msm8952", sizeof(hw_info->name));
     }  else if (!strcmp(snd_card_name, "msm8976-tasha-snd-card")) {
         strlcpy(hw_info->name, "msm8976", sizeof(hw_info->name));
+    }  else if (!strcmp(snd_card_name, "sdm660-snd-card")) {
+        strlcpy(hw_info->name, "sdm660", sizeof(hw_info->name));
+    }  else if (!strcmp(snd_card_name, "sdm660-tasha-snd-card")) {
+        strlcpy(hw_info->name, "sdm660", sizeof(hw_info->name));
     } else {
-        ALOGW("%s: Not an  8x16/8909/8952 device", __func__);
+        ALOGW("%s: Not an  8x16/8909/8952/8976/sdm660 device", __func__);
     }
 }
 
@@ -70,7 +74,8 @@ void *hw_info_init(const char *snd_card_name)
     }
 
     if (strstr(snd_card_name, "msm8x16") || strstr(snd_card_name, "msm8909")
-        || strstr(snd_card_name, "msm8952") || strstr(snd_card_name, "msm8976")) {
+        || strstr(snd_card_name, "msm8952") || strstr(snd_card_name, "msm8976")
+        || strstr(snd_card_name, "sdm660")) {
         ALOGV("8x16 - variant soundcard");
 
         strlcpy(hw_info->type, "", sizeof(hw_info->type));
-- 
2.13.0

